doc/doc-requirements.txt

```diff
@@ -7,3 +7,4 @@ memory_profiler
 pytest
 PyGitHub
 pydata_sphinx_theme
+myst_parser
```

Comment:
From the RTD build, it looks like sphinx < 2 is used:
![sphinx-version](https://user-images.githubusercontent.com/7579677/87770318-90fcfc80-c827-11ea-9eae-cb99854e29bb.png)

`myst-parser` requires sphinx > 2 (ref https://github.com/executablebooks/MyST-Parser/blob/master/setup.py#L44),
which makes me think that the error might be related with this version mismatch. We ran into similar issues in this repos:
* https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1658
* https://github.com/jupyterhub/dockerspawner/pull/378

@consideRatio comment here does a great job explaining why this happens.

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/bef1eaa6d23ad835185a675b9847585b87d87b60/doc/doc-requirements.txt#L6-L9

Reply:
Yepp I think this is one of these very frustrating issues I've run into in the past. ReadTheDocs (RTD) is a bit tricky as their python environment come preinstalled with an old version (1.8.5) of sphinx, and we want sphinx >2,<3 for myst-parser.

Putting myst-paser in the top of the doc-requirements.txt probably resolves this. Thank you @GeorgianaElena for such a great writeup with references etc!!

Reply:
Action points in doc-requirements.txt
- Move myst-parser[sphinx] to the top of the doc-requirements.txt file
- Remove sphinx (part of myst-parser[sphinx] with correct version constraints etc.
- Remove recomonmark

Reply:
Wow thanks so much for helping me troubleshoot this one!  I was stuck for several hours trying to debug this.  You folks are a life saver

Reply:
this is partially MyST's fault because we haven't yet added support for Sphinx 3, sorry 😞

Reply:
Awww =/ Yikes yeah I also got stuck on this several hours, sorry you had to experience this as well! Thanks for your work on this @hamelsmu! :tada: